src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -41,7 +41,7 @@ public virtual void Uninitialize()
             _features.Fetch(ref _features.Cache.Query, f => new QueryFeature(f));
 
         private IFormFeature FormFeature =>
-            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f));
+            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f, (f._context as DefaultHttpContext)?.FormOptions));
```

Comment:
I mean pushing it to ctor's of `DefaultHttpRequest` (and `DefaultHttpResponse`):
```csharp
private readonly FormOptions _formOptions;

public DefaultHttpRequest(HttpContext context, FormOptions formOptions)
{
    Initialize(context);
    _formOptions = formOptions;
}

// ...

private IFormFeature FormFeature => _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f, f._formOptions);
```

And change initializing `DefaultHttpContext` to:
```csharp
protected virtual HttpRequest InitializeHttpRequest() => new DefaultHttpRequest(this, FormOptions);
```